csrc/quantization/cutlass_w8a8/scaled_mm_c3x.cu

```diff
@@ -278,6 +278,80 @@ struct sm90_fp8_config<InType, OutType, Epilogue, 64> {
                       KernelSchedule, EpilogueSchedule>;
 };
 
+template <typename InType, typename OutType,
+          template <typename, typename, typename> typename Epilogue, int32_t M,
+          bool IsSmallN>  // IsSmallN is true if N < 8192
+struct sm90_int8_config {
+  static_assert(std::is_same<InType, int8_t>());
+  using KernelSchedule =
+      typename cutlass::gemm::KernelTmaWarpSpecializedPingpong;
+  using EpilogueSchedule = typename cutlass::epilogue::TmaWarpSpecialized;
+  using TileShape = Shape<_128, _128, _128>;
+  using ClusterShape = Shape<_2, _1, _1>;
+  using Cutlass3xGemm =
+      cutlass_3x_gemm<InType, OutType, Epilogue, TileShape, ClusterShape,
+                      KernelSchedule, EpilogueSchedule>;
+};
+
+template <typename InType, typename OutType,
+          template <typename, typename, typename> typename Epilogue,
+          bool IsSmallN>
+struct sm90_int8_config<InType, OutType, Epilogue, 128, IsSmallN> {
+  // Specialization for M in (64, 128] and any N
+  static_assert(std::is_same<InType, int8_t>());
+  using KernelSchedule =
+      typename cutlass::gemm::KernelTmaWarpSpecializedPingpong;
+  using EpilogueSchedule = typename cutlass::epilogue::TmaWarpSpecialized;
+  using TileShape = Shape<_64, _128, _128>;
+  using ClusterShape = Shape<_2, _1, _1>;
+  using Cutlass3xGemm =
+      cutlass_3x_gemm<InType, OutType, Epilogue, TileShape, ClusterShape,
+                      KernelSchedule, EpilogueSchedule>;
+};
```

Comment:
One high level comment about these PRs: I think the template specialization makes it a little less clear what's going on than just having a different function name for each case, especially because the template parameters aren't really functional -- the functions never actually look at `IsSmallN` and `M`

This is a little bit of a nit, however, as the dispatching in these kernels is quite a few levels deep at this point and I think we'll want to clean it up anyway in a separate PR after this and https://github.com/vllm-project/vllm/pull/5560 land

Reply:
Thanks @tlrmchlsmth . 
> I think the template specialization makes it a little less clear what's going on than just having a different function name for each case
I agree. I'll take a stab at making it better 👍 

> This is a little bit of a nit, however, as the dispatching in these kernels is quite a few levels deep at this point and I think we'll want to clean it up 
Yes. I have a plan, 
  - we can move the `cutlass_2x_gemm/cutlass_3x_gemm` structs to a `.cuh`, and
  - move the `[arch]_config` structs and the corresponding dispatch function into a `.cuh`, then 
  - simply call right thing from the `.cu`